// angular2-signaturepad/signature-pad.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'angular2-signaturepad/signature-pad';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
var renderType_SignaturePad_Host = null;
var _View_SignaturePad_Host0 = (function (_super) {
    __extends(_View_SignaturePad_Host0, _super);
    function _View_SignaturePad_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_SignaturePad_Host0, renderType_SignaturePad_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_SignaturePad_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('signature-pad', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_SignaturePad0(this.viewUtils, this.injector(0), this._appEl_0);
        this._SignaturePad_0_4 = new import3.SignaturePad(new import8.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._SignaturePad_0_4, [], compView_0);
        compView_0.create(this._SignaturePad_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_SignaturePad_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.SignaturePad) && (0 === requestNodeIndex))) {
            return this._SignaturePad_0_4;
        }
        return notFoundResult;
    };
    _View_SignaturePad_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._SignaturePad_0_4.ngAfterContentInit();
            }
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_SignaturePad_Host0;
}(import1.AppView));
function viewFactory_SignaturePad_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_SignaturePad_Host === null)) {
        (renderType_SignaturePad_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_SignaturePad_Host0(viewUtils, parentInjector, declarationEl);
}
export var SignaturePadNgFactory = new import10.ComponentFactory('signature-pad', viewFactory_SignaturePad_Host0, import3.SignaturePad);
var styles_SignaturePad = [];
var renderType_SignaturePad = null;
var _View_SignaturePad0 = (function (_super) {
    __extends(_View_SignaturePad0, _super);
    function _View_SignaturePad0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_SignaturePad0, renderType_SignaturePad, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_SignaturePad0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'canvas', null);
        this.init([], [this._el_0], [], []);
        return null;
    };
    return _View_SignaturePad0;
}(import1.AppView));
export function viewFactory_SignaturePad0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_SignaturePad === null)) {
        (renderType_SignaturePad = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, styles_SignaturePad, {}));
    }
    return new _View_SignaturePad0(viewUtils, parentInjector, declarationEl);
}
